src/Scout/ScoutEngine.php

```diff
@@ -435,14 +437,16 @@ public function createIndex($name, array $options = []): void
     {
         assert(is_string($name), new TypeError(sprintf('Argument #1 ($name) must be of type string, %s given', get_debug_type($name))));
 
+        $definition = $this->indexDefinitions[$name] ?? self::DEFAULT_DEFINITION;
+        if (! isset($definition['mappings'])) {
+            throw new LogicException(sprintf('The search index definition for collection "scout.mongodb.index-definition.%s" must contain a "mappings" key. Find documentation at https://www.mongodb.com/docs/atlas/atlas-search/create-index/', $name));
```

Comment:
I'm not sure about adding a link here. Can be useful, maybe update it when the laravel docs is written.

Reply:
Would InvalidArgumentException be preferable, or is there prior art for using LogicException directly?

https://www.mongodb.com/docs/atlas/atlas-search/create-index/ seems fine if you're in the habit of adding doc links to exception messages.

https://www.mongodb.com/docs/manual/reference/command/createSearchIndexes/#search-index-definition-syntax may be a more direct reference, though.

Reply:
I updated the exception class and the message with your link.